File: src/testNorm.c

#include <R_ext/Lapack.h>
#include <R_ext/Applic.h>
#include <R_ext/PrtUtil.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>
#include <R.h>
#include <Rmath.h>
#include "utility.h"



/*********************************************************************
 *
 * 
 *
 *********************************************************************/
 
void testNormC(double* Normoutput,int* dims)
{
    int nnorm = dims[0], j;
    GetRNGstate();
    for(j=0; j<nnorm; j++){
      Normoutput[j] = rnorm(0,1);
    }
    PutRNGstate();
} 

File: R/testNorm.R
`testNorm` <- function(nnorm = 100000){


  dims = rep(0,1)
  dims[1] = nnorm
  
 
  Normoutput = rep(0,nnorm)

  
  Z = .C("testNormC", Normoutput=as.double(Normoutput),as.integer(dims), PACKAGE="SubsetRV")

  Normoutput= Z$Normoutput
  return(Normoutput)
}

